platforms/core-runtime/internal-instrumentation-processor/src/main/java/org/gradle/internal/instrumentation/extensions/property/PropertyUpgradeAnnotatedMethodReader.java

```diff
@@ -146,6 +157,13 @@ public Collection<Result> readRequest(ExecutableElement method) {
 
     @SuppressWarnings("unchecked")
     private List<AccessorSpec> readAccessorSpecsFromReplacesEagerProperty(ExecutableElement method, AnnotationMirror annotationMirror) {
+        Element element = AnnotationUtils.findAnnotationValueWithDefaults(elements, annotationMirror, "adapter")
+            .map(v -> types.asElement((TypeMirror) v.getValue()))
+            .orElseThrow(() -> new IllegalArgumentException("Missing adapter value"));
+        if (!element.getSimpleName().toString().equals(DefaultValue.class.getSimpleName())) {
+            return readAccessorSpecsFromAdapter(element, method.getEnclosingElement(), annotationMirror);
```

Comment:
💭 Here we read an adapter from annotation and then in `readAccessorSpecsFromAdapter` read methods from adapter